adapters/thetradedesk/thetradedesk_test.go

```diff
@@ -0,0 +1,385 @@
+package thetradedesk
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/adapters/adapterstest"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEndpointTemplateMalformed(t *testing.T) {
+	_, buildErr := Builder(openrtb_ext.BidderTheTradeDesk, config.Adapter{
+		Endpoint: "{{Malformed}}"},
+		config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+
+	assert.Error(t, buildErr)
+}
+
+func TestBadConfig(t *testing.T) {
+	_, buildErr := Builder(openrtb_ext.BidderTheTradeDesk, config.Adapter{
+		Endpoint:         `http://it.doesnt.matter/bid`,
+		ExtraAdapterInfo: "12365217635",
+	},
+		config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+
+	assert.Error(t, buildErr)
+}
+
+func TestCorrectConfig(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderTheTradeDesk, config.Adapter{
+		Endpoint:         `http://it.doesnt.matter/bid`,
+		ExtraAdapterInfo: `abcde`,
+	},
+		config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+
+	assert.NoError(t, buildErr)
+	assert.NotNil(t, bidder)
+}
+
+func TestEmptyConfig(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderTheTradeDesk, config.Adapter{
+		Endpoint:         `https://direct.adsrvr.org/bid/bidder/{{.SupplyId}}`,
+		ExtraAdapterInfo: `ttd`,
+	},
+		config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+
+	assert.NoError(t, buildErr)
+	assert.NotNil(t, bidder)
+}
+
+func TestJsonSamples(t *testing.T) {
+	bidder, err := Builder(
+		openrtb_ext.BidderTheTradeDesk,
+		config.Adapter{Endpoint: "https://direct.adsrvr.org/bid/bidder/{{.SupplyId}}", ExtraAdapterInfo: "ttd"},
+		config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "1"},
+	)
+	assert.Nil(t, err)
+	adapterstest.RunJSONBidderTest(t, "thetradedesktest", bidder)
+}
+
+func TestGetBidType(t *testing.T) {
+	type args struct {
+		markupType openrtb2.MarkupType
+	}
+	tests := []struct {
+		name              string
+		args              args
+		markupType        openrtb2.MarkupType
+		expectedBidTypeId openrtb_ext.BidType
+		wantErr           bool
+	}{
+		{
+			name: "getBidType banner",
```

Comment:
You can simplify the names of your tests by removing `getBidType` from the names (e.g. `name: "banner"`) since the test name `TestGetBidType` already makes it obvious what is under test and any test failure will you give you the test name `TestGetBidType` along with the test case name.